packages/fury-adapter-oas3-parser/lib/parser/oas/parseReferenceObject.js

```diff
@@ -52,23 +52,22 @@ function parseReferenceObject(context, componentName, element, returnReferenceEl
       return createError(namespace, `'#/components/${componentName}' is not defined`, ref);
     }
 
-    const element = component.getMember(referenceParts[3]);
-    if (!element) {
+    const componentId = referenceParts[3];
+    if (!component.hasKey(componentId)) {
       return createError(namespace, `'${ref.toValue()}' is not defined`, ref);
     }
 
     if (returnReferenceElement) {
       const element = new context.namespace.elements.Element();
-      const id = referenceParts[3];
-      element.element = id;
+      element.element = componentId;
       return element;
     }
 
-    if (element.value) {
-      return element.value;
-    }
-
-    return new namespace.elements.ParseResult([]);
+    return new namespace.elements.ParseResult(
+      component
+        .filter((value, key) => key.toValue() === componentId && value)
+        .map(value => value)
```

Comment:
What is the reason for this line?

Reply:
I spent some fun fighting the differences between filter/map on an ObjectSlice(ObjectElement) while writing this code. Filter here is returning an ObjectSlice of Member Elements, however the map transform function signature is expanded to be (`value`, `key`, `member`) and thus I want to map the member to value, value is the first argument so it is just passing it back through.

Reply:
I was suspecting something like this, but better asked :)